lnwallet/chanfunding/canned_assembler.go

```diff
@@ -171,13 +189,20 @@ type CannedAssembler struct {
 	// a normal channel. Until this height, it's considered frozen, so it
 	// can only be cooperatively closed by the responding party.
 	thawHeight uint32
+
+	// musig2 determines if the funding output should use musig2 to
+	// generate an aggregate key to use as the taproot-native multi-sig
+	// output.
+	musig2 bool
 }
 
 // NewCannedAssembler creates a new CannedAssembler from the material required
 // to construct a funding output and channel point.
+//
+// TODO(roasbeef): pass in chan type instead?
```

Comment:
yeah was gonna ask about this (switching on chan type everywhere in this commit instead of on `musig2`) cause is there not a possible future where we use `musig2` for the funding output but with a slightly different structure? for example in a future where we dont include the bip86 tweak for w/e reason

Reply:
Yep good point, will go with the chan type here everywhere instead.